sw/device/lib/base/hardened.h

```diff
@@ -552,12 +559,19 @@ inline uintptr_t ct_cmovw(ct_boolw_t c, uintptr_t a, uintptr_t b) {
 #define HARDENED_CHECK_OP_GE_ "bgeu"
 
 // clang-format off
-#define HARDENED_CHECK_(op_, a_, b_) \
-  asm volatile(                      \
-      op_ " %0, %1, .L_HARDENED_%=;" \
-      HARDENED_UNIMP_SEQUENCE_()     \
-      ".L_HARDENED_%=:;"             \
-      ::"r"(a_), "r"(b_))
+#define HARDENED_CHECK_(c_op_, op_, a_, b_)       \
+  do {                                            \
+    asm volatile(                                 \
+      op_ " %0, %1, .L_HARDENED_%=;"              \
+      HARDENED_UNIMP_SEQUENCE_()                  \
+      ".L_HARDENED_%=:;"                          \
+      ::"r"(a_), "r"(b_));                        \
+    /* Demonstrate that code after the  unimp     \
+     * sequence is unreachable. */                \
+    if (! ((a_) c_op_ (b_))) {                    \
+      OT_UNREACHABLE();                           \
+    }                                             \
```

Comment:
this can affect the size of the ROM. Can you check the disassembly before and after this change?

For checking the disassembly you can use:
`./bazelisk.sh build-then "less %s" //sw/device/silicon_creator/rom:rom_with_fake_keys_sim_dv_dis`

and for the size report:
`python3 -m util.py.scripts.object_size --configs riscv32 //sw/device/silicon_creator/rom:rom_with_fake_keys_sim_dv`

Reply:
I peeked at a few places and it doesn't appear to change the generated code. For instance, I checked [rom.c](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom.c;l=325;drc=852a34e6ed7651f7e894314fdf33d56f015ad297) and it's the same code generated before and after.

I can't explain this, but the change does increase binary size by a few bytes.

```diff
diff --git a/object_size_before.txt b/object_size.txt
index e7eadb567..03a5575d6 100644
--- a/object_size_before.txt
+++ b/object_size.txt
@@ -5,10 +5,10 @@
     bazel-out/k8-fastbuild/bin/sw/device/silicon_creator/rom/rom_with_fake_keys_sim_dv.elf



   File size:

-    550080 bytes (537.2 KiB)

+    552200 bytes (539.3 KiB)



   Memory utilization overview:

-    rom                  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━        86% Available: 4262 bytes (4.2 KiB), Used: 28506 bytes (27.8 KiB) of 32768 bytes (32.0 KiB)

+    rom                  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸       87% Available: 4254 bytes (4.2 KiB), Used: 28514 bytes (27.8 KiB) of 32768 bytes (32.0 KiB)

     ram_main             ━━╸                                        7% Available: 121756 bytes (118.9 KiB), Used: 9316 bytes (9.1 KiB) of 131072 bytes (128.0 KiB)

     eflash                                                          0% Available: 1048576 bytes (1024.0 KiB), Used: 0 bytes (0.0 KiB) of 1048576 bytes (1024.0 KiB)

     ram_ret_aon                                                     0% Available: 4096 bytes (4.0 KiB), Used: 0 bytes (0.0 KiB) of 4096 bytes (4.0 KiB)

@@ -22,8 +22,8 @@
   ┡━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━┩

   │         .vectors │   rom    │    384 bytes (0.4 KiB) │ VMA: 00008000, LMA: 00008000     │ 256 bytes │ SHT_PROGBITS │   0x6 │

   │             .crt │   rom    │    950 bytes (0.9 KiB) │ VMA: 00008180, LMA: 00008180     │   4 bytes │ SHT_PROGBITS │   0x6 │

-  │            .text │   rom    │ 24320 bytes (23.8 KiB) │ VMA: 00008538, LMA: 00008538     │   4 bytes │ SHT_PROGBITS │   0x6 │

-  │        .shutdown │   rom    │    792 bytes (0.8 KiB) │ VMA: 0000e438, LMA: 0000e438     │   4 bytes │ SHT_PROGBITS │   0x6 │

+  │            .text │   rom    │ 24328 bytes (23.8 KiB) │ VMA: 00008538, LMA: 00008538     │   4 bytes │ SHT_PROGBITS │   0x6 │

+  │        .shutdown │   rom    │    792 bytes (0.8 KiB) │ VMA: 0000e440, LMA: 0000e440     │   4 bytes │ SHT_PROGBITS │   0x6 │

   │          .rodata │   rom    │   3100 bytes (3.0 KiB) │ VMA: 0000e760, LMA: 0000e760     │  32 bytes │ SHT_PROGBITS │   0x2 │

   │       .chip_info │   rom    │      8 bytes (0.0 KiB) │ VMA: 0000ff80, LMA: 0000ff80     │   8 bytes │ SHT_PROGBITS │   0x2 │

   │ .static_critical │ ram_main │   8132 bytes (7.9 KiB) │ VMA: 10000000, LMA: N/A (NOBITS) │   4 bytes │  SHT_NOBITS  │   0x3 │
```